launch/launch/actions/execute_process.py

```diff
@@ -378,10 +379,19 @@ def get_sub_entities(self):
             return self.__on_exit
         return []
 
+    def __deferred_shutdown(self, event, context):
+        return self._shutdown_process(context, send_sigint=True)
+
     def _shutdown_process(self, context, *, send_sigint):
         if self.__shutdown_received:
             # Do not handle shutdown more than once.
             return None
+
+        if self.process_details is None or self._subprocess_transport is None:
+            # Skip shutting down if the process is not ready
+            context.register_event_handler(OnProcessStart(on_start=self.__deferred_shutdown))
```

Comment:
@Blast545 how do you prevent this event handler being added multiple times?

Reply:
@hidmic The handler itself might be added multiple times if the shutdown process is called multiple times before the subprocess_transport is available.

However, the first check of the function `if self.__shutdown_received:` doesn't allow for the shutdown actions to be added to the actions queue multiple times.